pkg/config/application.go

```diff
@@ -396,6 +406,12 @@ type WaitApprovalStageOptions struct {
 	MinApproverNum int      `json:"minApproverNum" default:"1"`
 }
 
+type CustomSyncOptions struct {
```

Comment:
Should define validate function for this StageCustomSync and add it here
https://github.com/pipe-cd/pipecd/blob/master/pkg/config/application.go#L135

Reply:
https://github.com/pipe-cd/pipecd/pull/4223#discussion_r1129136931